src/app/components/dashboard/dashboard.component.html

```diff
@@ -731,21 +731,36 @@
 			</ion-label>
 		</ion-item>
 
-		<ion-item
-			lines="none"
-			*ngIf="data.foreignValidator"
-			tooltip="0x00 credentials must be upgraded to 0x01 credentials at some point in time to be eligible for withdrawals."
-			placement="right"
-			trigger="click"
-			hideDelayAfterClick="5000">
-			<ion-label class="stat-title"> Withdrawal Cred </ion-label>
-			<ion-label
-				class="value"
-				[class]="(data.foreignValidatorItem.data.withdrawalcredentials | slice : 0 : 4) == '0x01' ? 'ok-proposals' : 'warn'"
-				(click)="switchCurrencyPipe()">
-				{{ data.foreignValidatorItem.data.withdrawalcredentials | slice : 0 : 4 }}
-			</ion-label>
-		</ion-item>
+		<span *ngIf="data.foreignValidator">
+			<div *ngIf="(data.foreignValidatorItem.data.withdrawalcredentials | slice : 0 : 4) == '0x01'; then creds0x01; else creds0x00"></div>
+			<ng-template #creds0x01>
+				<ion-item
+					lines="none"
+					tooltip="This validator has 0x01 credentials and is ready for withdrawals."
+					placement="right"
+					trigger="click"
+					hideDelayAfterClick="5000">
+					<ion-label class="stat-title"> Withdrawal Cred </ion-label>
+					<ion-label class="value" [class]="'ok - proposals'" (click)="switchCurrencyPipe()">
+						{{ data.foreignValidatorItem.data.withdrawalcredentials | slice : 0 : 4 }}
+					</ion-label>
+				</ion-item>
+			</ng-template>
+
+			<ng-template #creds0x00>
+				<ion-item
+					lines="none"
+					tooltip="0x00 credentials must be upgraded to 0x01 credentials at some point in time to be eligible for withdrawals."
+					placement="right"
+					trigger="click"
+					hideDelayAfterClick="5000">
+					<ion-label class="stat-title"> Withdrawal Cred </ion-label>
+					<ion-label class="value" [class]="'warn'" (click)="switchCurrencyPipe()">
+						{{ data.foreignValidatorItem.data.withdrawalcredentials | slice : 0 : 4 }}
+					</ion-label>
+				</ion-item>
+			</ng-template>
+		</span>
```

Comment:
You could define a variable for the tooltip and override it in the code. 
`tooltip="{{ withdrawalCredentialTooltip }}"`

or maybe make it boolean so you can toggle the color there as well and show the relevant text via ternary.

`(click)="switchCurrencyPipe()"` should not be here, we usually only use this on currency values so the user can switch back and forth. That's on me though since it was already here in the first place. Best to remove it

Reply:
Good call, I used the "Boolean+Ternary"-approach.

The boolean is stored in `OverviewData` here:
https://github.com/gobitfly/eth2-beaconchain-explorer-app/blob/8569796c5673869d9afb7bd64dd3deecfcafdf01/src/app/controllers/OverviewController.ts#L49

One might argue that it should be put directly in `ValidatorResponse` here:
https://github.com/gobitfly/eth2-beaconchain-explorer-app/blob/e15daac4f7cb85a589b0c8d533d7ae981f9be566/src/app/requests/requests.ts#L179

But this would require us to either change the end-point in the Explorer or to set the data in the app manually whenever a `ValidatorResponse` is received. That felt like too big of a change for this bugfix. What do you think?

Reply:
I think the proposed solution is good enough. In theory you could also write a method in OverviewController `isWithdrawalCredential0x1(data.foreignValidatorItem.data.withdrawalcredentials)` that returns bool based in what is in the response from the server and use that in the ternary. A pipe would also work but given we probably only need it here once a method is probably better, pipes are better for stuff we use often.